Documentation/dotnet-primer.md

```diff
@@ -0,0 +1,147 @@
+
+.NET languages can be object-oriented, with hiearchies of base and derived classes. The .NET runtime will only allow object casts and calls that align with the object hierarchy.
+
+	Dog dog = Dog.AdoptDog();
+	Pet pet = (Pet)dog; // Dog derives from Pet
+	pet.ActCute(); 
+	Car car = (Car)dog; // will throw - no relationship between Car and Dog
+	object temp = (object)dog; // legal - a Dog is an object
+	car = (Car)temp; // will throw - the runtime isn't fooled
+	car.Accelerate() // the dog won't like this, nor will the program get this far
+
+Type safety also guarantees the fidelity of accessor keywords (e.g. private, public, internal). This is particularly useful for non-public data that an implementation uses to manage its behavior.
+
+	Dog dog = Dog._nextDogToBeAdopted; // will throw - this is a private field
+
+Generic Types
+-------------
+
+Generic types enable a kind of uncoordinated mix-ins of (largely) arbitrary types. They allow the creation of a kind of _compound type_. A generic type is considered to be _open_, as it requires a set of types (however many type parameters it exposes) to be provided in order to be _closed_. While static in nature, like the rest of the .NET type system, generics provide a kind of faux dynamism that developers find very useful.
+
+The most commonly used generic type is `List<T>` and its sibling `Dictionary<K,V>`. You can use these types, and also create your own generic types. The following example demonstrates the use of these two built-in generic types.
+
+	List<string> strings = new List<string>();
+	strings.Add("string A");
+	strings.Add("string B");
+
+	Dictionary<int,List<string>> lists = new Dictionary<int,List<string>>();
+	lists.Add(DateTime.Now.Hour,strings); // new strings added every hour
+
+Async Programming
+-----------------
+
+Async is a first-class concept within .NET, with async support in the runtime, the framework libraries and various .NET languages. Async is based off of the `Task` concept, which encaptsulates a set of operations to be completed. Tasks are distinct from threads and may not rely on threads or require CPU time much at all, particularly for i/o-bound tasks.
+
+C# includes special treatment for async, including the special keyword `await` for managing tasks. The following example demonstrates calling a web service as an async operation.
+
+	string url = "http://someUrl";
+	HttpClient client = new HttpClient();
+	string json = await client.GetStringAsync(url);
+
+The call to `client.GetStringAsync(url)` does not block, but instead immediately yields by returing a `Task`. Computation resumes and the call returns the requested string when the network activity has completed.
```

Comment:
~~returing~~ returning.

I know this should just be a short introduction, but there's quite a lot happening here that might warrant another sentence or two.